### DISCUSSION TITLE
Lifecycle markdown

### DISCUSSION THREAD

**Discussion by yanivpaz** [rook/rook#9742] [Ideas]
Is there any file containing the versions which are not supported anymore ?
For example - are there still security fixes for v.1.5.7 ?

**Comment by leseb** [comment]
We don't have a strict public policy for this, it's probably a good opportunity to clarify this. We can certainly build a table and explicitly describe when a release goes EOL.
Typically, we do support up to 2 releases at the same time. For instance:

* today 1.8.x is the latest so it is obviously supported
* we still do patches for 1.7.x
* very rarely do patches for 1.6.x, I think it's fair to say 1.6 is EOL

Rook is a fast-moving project and in general, we strongly encourage users to upgrade to get the best of it.
To answer your question, we don't do security fixes on 1.5.x unless there is something that prevents you from upgrading and you are in a catastrophic situation.

**Reply by yanivpaz** [reply]
Thanks a lot for the answer . is there ETA for 1.7  EOL ?

**Reply by leseb** [reply]
Rook's major release schedule is every quarter, this time we will wait for Ceph Quincy to be out to release v1.9.0 so it might slip a little. Hopefully around April, then we can still support v1.7 for a few months, perhaps until June/July.

@travisn more thoughts on this?

**Reply by travisn** [reply]
I would hesitate to formalize the EOL for any specific release since we always do our best to support when there are critical issues. 
- We haven't had a history of requiring critical fixes on old releases as it's expected that upgrading to a more recent release is a reasonable way to get the critical fixes. 
- Starting in v1.6 we moved fully to github actions  for the CI and decommissioned Jenkins. The releases for v1.5 and older were all on Jenkins, so we should consider v1.5 and older officially EOL given the extra difficulty to release builds on those versions.
-If the community finds and fixes a critical bug fix that they want to backport, we would typically expect to merge it. If it's v1.5 or older, you would just need to run your own build. 
- If you want to support a specific rook version for longer than upstream expects, it would be recommended to create your own builds, likely from your own fork. This way you won't be dependent on what the upstream community decides. This is the pattern for downstream products that rely on Rook.
